core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -538,14 +538,17 @@ public String getDownloadUrl() {
 
     /**
      * Is this the legacy default update center site?
-     * @deprecated
-     *      Will be removed, currently returns always false.
-     * @since 2.343
+     * @since 1.357
      */
-    @Deprecated
     @Restricted(NoExternalUse.class)
     public boolean isLegacyDefault() {
-        return false;
+        return isJenkinsCI();
+    }
+
+    private boolean isJenkinsCI() {
+        return url != null
+                && UpdateCenter.PREDEFINED_UPDATE_SITE_ID.equals(id)
+                && url.startsWith("http://updates.jenkins-ci.org/");
```

Comment:
Adapting the pre-#6116 code to handle the migration from http://updates.jenkins-ci.org/ to https://updates.jenkins.io/ rather than the migration from http://hudson-ci.org to http://updates.jenkins-ci.org/